cmd/setup-buildpack-build/main.go

```diff
@@ -44,6 +44,11 @@ func main() {
 	w := os.Stdout
 
 	describe.SectionWriter(w, "Changing permissions", func(w io.Writer) {
+		fmt.Fprintf(w, "chmod -R %d %s\n", 0755, *workspace)
```

Comment:
I don't have enough context to say.

Reply:
Cloud Foundry uses a different UID as well, so I think we'll still be incompatible with them

Reply:
What UID does CF use?

Reply:
In the docker image `cloudfoundry/cnb`

```
cat /etc/passwd | grep vcap
vcap:x:2000:2000::/home/vcap:/bin/bash
```

Seems they use 200 with the name vcap

Reply:
Also `heroku/buildpacks` uses `heroku` ans `1000`

Reply:
Seems like we could offer some feedback to the buildpacks team about that.

Reply:
According to this: https://github.com/buildpack/pack/commit/5a87cc9b68fad3d9841b0f9c95f4cf7d0fa5a720
the lifecycle binaries *should* do the chowning if they run as root?

Reply:
And the uid can be a flag passed in to the analyzer. 
https://github.com/buildpack/lifecycle/blob/c829fcf064ed2fba4baaab585e45e8ac377501b9/cmd/flags.go#L112

I've opened up () to fix this in a future change

Reply:
We do support the flag and param already, however the CNB spec says the UID and GID are set as labels on the image, so we're supposed to fetch them off the image before trying to build--which we don't currently do.

If I had to guess, I bet CF uses the UID/GID=2000 as a way to enforce build safety, the main build process could run as root or UID=1000 and the user's source code gets UID=2000 to prevent them from reading creds.